example/lib/extensions.dart

```diff
@@ -0,0 +1,16 @@
+/// Extensions for the `String` class.
+extension StringX on String {
+  /// Converts the first character of the string to uppercase.
+  ///
+  /// If the string is empty, returns an empty string.
+  ///
+  /// If the string has only one character, returns the uppercase version of
+  /// the character.
+  ///
+  /// Otherwise, returns the string with the first character converted to
+  String toTitleCase() {
```

Comment:
By definition Title case capitalizes major words, such as nouns, pronouns, verbs, adjectives, adverbs, and proper nouns.
I think this is Sentence case